plc4j/api/src/main/java/org/apache/plc4x/java/api/value/PlcBYTE.java

```diff
@@ -181,6 +205,90 @@ public short getShort() {
         return value;
     }
 
+    @Override
+    @JsonIgnore
+    public boolean isInteger() {
+        return true;
+    }
+
+    @Override
+    @JsonIgnore
+    public int getInteger() {
+        return value.intValue();
+    }
+
```

Comment:
Instead of adding these for every type, wouldn't it make sense to have something like a "PlcIECBitStringValue", "PlcIECIntegerValue", ...?

Reply:
But thinking of it ... perhaps making it generic and passing in the type would help ...

Reply:
I think there a at least two use cases for these methods. To use them when parsing the value within the dataio classes. The other is by the user when they need to parse the value to a specific type. 

A generic function to be used when parsing like getValue() would be good.

For the user the getInteger, getByte, etc.. might be good as I can't think of an easy way to pass a type to the function and have it return the data type requested without just putting the existing functions within a switch statement.